File: src/models/Utilizador.java
package models;

public class Utilizador {
    private String nome;
    private String password;


    // Construtores
    public Utilizador() {
        this.nome = "";
        this.password = "";
    }

    public Utilizador(String nome, String password) {
        this.nome = nome;
        this.password = password;
    }

    public Utilizador(Utilizador outro) {
        this.nome = outro.nome;
        this.password = outro.password;
    }

    // Getters e Setters
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    @Override
    public Utilizador clone() {
        return new Utilizador(this);
    }

    @Override
    public String toString() {
        return "Utilizador{" +
                "nome='" + nome + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}

File: Python/backend/models/Utilizador.py
class Utilizador:
    def __init__(self, nome="", password="", outro=None):
        """
        Construtor da classe Utilizador
        Pode ser usado como:
        - Utilizador() - construtor padrão
        - Utilizador(nome, password) - construtor com parâmetros
        - Utilizador(outro=outro_utilizador) - construtor de cópia
        """
        if outro is not None:
            # Contrutor de cópia
            self.nome = outro.nome
            self.password = outro.password
        else:
            # Construtor padrão ou com parâmetros
            self.nome = nome
            self.password = password
            
    # Getters e Setters
    def get_nome(self):
        return self.nome
    
    def set_nome(self, nome):
        self.nome = nome
        
    def get_password(self):
        return self.password
    
    def set_password(self, password):
        self.password = password
        
    def clone(self):
        return Utilizador(outro=self)
    